src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java

```diff
@@ -572,13 +572,47 @@ public int nextCycle(int cycle, @NotNull TailerDirection direction) throws Parse
         return pool.nextCycle(cycle, direction);
     }
 
+    @Deprecated
+    /**
+     * Instead, {@link #approximateExcerptsInCycle(int)} or {@link #exactExcerptsInCycle(int)} should be called.
+     */
     public long exceptsPerCycle(int cycle) {
+        return approximateExcerptsInCycle(cycle);
+    }
+
+    /**
+     * Returns a number of excerpts in a cycle. May use a fast path to return the cycle length cached in indexing,
+     * which is updated last during append operation so may be possible that a single entry is available for reading
+     * but not acknowledged by this method yet.
+     */
+    public long approximateExcerptsInCycle(int cycle) {
+        throwExceptionIfClosed();
+        // TODO: this function may require some re-work now that acquireTailer has been deprecated
+        StoreTailer tailer = acquireTailer();
+        try {
+            if (tailer.moveToCycle(cycle)) {
```

Comment:
consider -> return tailer.moveToCycle(cycle) ? tailer.store.approximateLastSequenceNumber(tailer) + 1 : -1;

Reply:
which should be the same thing